src/framework/ui/list/listItem.component.tsx

```diff
@@ -0,0 +1,221 @@
+
+    return (
+      <Text
+        style={[style, styles.title]}
+        key={2}>
+        {title}
+      </Text>
+    );
+  };
+
+  private renderDescriptionElement = (style: StyleType): React.ReactElement<TextProps> => {
+    // @ts-ignore: will be not executed if `description` property is provided
+    const { description } = this.props;
+
+    return (
+      <Text
+        key={3}
+        style={[style, styles.description]}>
+        {description}
+      </Text>
+    );
+  };
+
+  private renderAccessoryElement = (style: StyleType): React.ReactElement<any> => {
+    // @ts-ignore: will be not executed if `accessory` property is provided
+    const { index, accessory } = this.props;
+
+    const accessoryElement: React.ReactElement<any> = accessory(index, style);
+
+    return React.cloneElement(accessoryElement, {
+      key: 4,
+      style: [accessoryElement.props.style, style, styles.accessory],
+    });
+  };
+
+  private renderContentElementChildren = (style: StyleType): React.ReactNode => {
+    // @ts-ignore: will be not executed if any of properties below is provided
+    const { title, description } = this.props;
+
+    const hasTitle: boolean = title !== undefined;
+    const hasDescription: boolean = description !== undefined;
+
+    return [
+      hasTitle ? this.renderTitleElement(style.title) : undefined,
+      hasDescription ? this.renderDescriptionElement(style.description) : undefined,
+    ];
+  };
+
+  private renderTemplateChildren = (style: StyleType): React.ReactNode => {
+    // @ts-ignore: following props could not be provided
+    const { icon, title, description, accessory } = this.props;
+
+    const hasIcon: boolean = icon !== undefined;
+    const hasContent: boolean = title !== undefined || description !== undefined;
+    const hasAccessory: boolean = accessory !== undefined;
+
+    return [
+      hasIcon ? this.renderIconElement(style.icon) : undefined,
+      hasContent ? this.renderContentElement(style) : undefined,
+      hasAccessory ? this.renderAccessoryElement(style.accessory) : undefined,
+    ];
+  };
+
+  public render(): React.ReactElement<TouchableOpacityProps> {
+    const { style, themedStyle, ...derivedProps } = this.props;
+    const { container, ...componentStyles } = this.getComponentStyle(themedStyle);
+    const componentChildren: React.ReactNode = this.renderTemplateChildren(componentStyles);
+
+    return (
+      <TouchableOpacity
+        {...derivedProps}
+        style={[container, style, styles.container]}
+        activeOpacity={1.0}
+        onPress={this.onPress}
+        onPressIn={this.onPressIn}
+        onPressOut={this.onPressOut}
+        onLongPress={this.onLongPress}>
+        {componentChildren}
+      </TouchableOpacity>
+    );
+  }
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flexDirection: 'row',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  contentContainer: {
+    flex: 1,
+  },
+  icon: {},
```

Comment:
Each component has 3 ways to be styled:
* Using `eva` mappings
* Using it's own strict style
* Using user-defined style

This just helps to support consistency because the resulting style property would be:
```
style=[evaStyle, userStyle, strictStyle]
```